### ISSUE TITLE
Feature request: separate black/white lists

### ISSUE THREAD

**Issue by ershov** [darkreader/darkreader#28]
Hi,
I love this extension and I have a proposal for improvement.
Today, there's only one list of sites which acts as 'inverted list' or 'no touch list' depending on the mode.
This doesn't make much sense because the sites that are chosen to be inverted should always be inverted, not depending on the default mode. And vice versa.
The suggestion is: there should be two separate lists for sites to be inverted and for sites to stay untouched. And the 'default' mode will do the default to the sites that are not in any of the lists.

**Comment by alexanderby** [comment]
@ershov This figure explains why one Site List covers all use cases.
![dark-reader_site-list_diagram](https://cloud.githubusercontent.com/assets/9638588/16033692/1fd826e8-3216-11e6-9f8b-6c7101d77a19.png)

**Comment by ershov** [comment]
I'm not quite sure what it explains.

The behavior that I see today is as follows:
1. Add an URL (e.g. google.com) to the 'sites list'
2. If 'not invert listed' selected, the site is not inverted.
3. If 'invert listed only' selected, the site is inverted.

Once I start using 'sites list', I can't switch the default mode.
The site 'google.com' is always white => I always want it inverted.
The site 'store.steampowered.com' is always dark => I never want it inverted.
But there are lots of unlisted sites which I want to see in the 'default' mode.

**Comment by alexanderby** [comment]
There are two use cases:
1. You want most of the sites to be dark. Then use "Not invert listed" for ignoring sites which are inverted incorrectly (e.g. already dark, but inverts into white).
2. You want few sites to be dark. Then use "Invert listed only" to specify what to invert.
These two strategies are illustrated at the bottom of the figure.

**Comment by ershov** [comment]
Right.
Here's the problem: I have to choose the mode from the very beginning and can't change the mode once I have any sites in the list.

**Comment by alexanderby** [comment]
I find this case very specific, and adding additional list will make UI more complex. So, I think it's more simple and universal to add a JSON config editor.

**Comment by RalphLesch** [comment]
Related: #147 
For e.g. subdomains cannot be fully controlled by the current include/exclude mechanics and URL syntax.
If not using both include and exclude lists, I would suggest changing the URL syntax to allow better control of the includes/excludes.

**Comment by alexanderby** [comment]
I was thinking of adding unlimited count of settings groups. Currently I'm refactoring the UI, so will work on it later.

**Comment by RalphLesch** [comment]
Sub groups for domains should be enough to solve most URL match problems.
I would suggest an extended hierarchy:
- common/shared group, with inversion fixes that are always used (unless deactivated by an option in the other rules maybe?).
- site / domain groups for URL pattern, will be used unless subdomain matches are found.
  - subdomain groups: overwrite (super)domain rules.

Then a layout like this is possible:
```json
{
    "common": { "invert etc (fixes)": "", 
    },
    "sites": {
        "*": {
            "fixes": "",
            "subsites": {
                "domain1-pattern": {
                    "fixes": "",
                    "subsites": {
                      "subdomain1-pattern": {},
                      "subdomain2-pattern": {}
                    }
                },
                "domain2-pattern": {}
            }
        }
    }
}
```
Side note: collapsible sub groups in the editor and/or a complete separation for the sites (show only site config) would help a lot in the new UI.

**Comment by elsaxo** [comment]
> Hi,
> I love this extension and I have a proposal for improvement.
> Today, there's only one list of sites which acts as 'inverted list' or 'no touch list' depending on the mode.
> This doesn't make much sense because the sites that are chosen to be inverted should always be inverted, not depending on the default mode. And vice versa.
> The suggestion is: there should be two separate lists for sites to be inverted and for sites to stay untouched. And the 'default' mode will do the default to the sites that are not in any of the lists.

**Comment by brendan8c** [comment]
I love this extension! But I also think that the lists need to be changed! Implementation at this stage is extremely inconvenient and impractical.
I am using dark mode. And I only need one list in which I will add sites. Sites that will be in this list will be inverted to dark mode. All other sites that have not been listed should not be affected by the extension.

**Comment by alexanderby** [comment]
Some ideas for the UI

<img width="304" alt="Screen Shot 2021-09-19 at 15 18 18" src="https://user-images.githubusercontent.com/9638588/133933031-5a7d99b7-6a73-47aa-87fe-c901629c8834.png">
<img width="300" alt="Screen Shot 2021-09-19 at 15 20 31" src="https://user-images.githubusercontent.com/9638588/133933034-9fd5704a-497b-4505-94dd-2327f8d19447.png">

**Comment by brendan8c** [comment]
I think we only need to have lists to include styles.
And for those sites that are not on the list, the extension will not respond.
Why do we need a list to disable styles?
If the styles will be applied only to those websites that are on the list.

**Comment by alexanderby** [comment]
Some users have the extension enabled by default, they will use a list for disabled styles. Some users have the extension disabled by default, they will use a list for enabled styles. Some users switch between these to modes, they will use both styles when they switch.

Another use case is, when user wants to disable `www.google.com`, but at the same time to enable `www.google.com/maps`.

**Comment by Faedelity** [comment]
I can see this being useful for someone who, for whatever reason, has to flip their default back and forth. I can see that being a thing for someone who sometimes has their own preference and sometimes has to use the other, such as on a work computer that is used for both work (where they might prefer default inversion) and presentations (where leaving things bright usually looks better). Not arguing either way on the feature, just this is a use case I have seen.

**Comment by thousandsofthem** [comment]
Should be easy to understand this way:

| default for unlisted sites  | [invert] |
| ------------- | ------------- |

| sites to always invert  | 
| ------------- |
| google.com  |
| ... |

| sites to never invert  | 
| ------------- |
| github.com  |
| ... |

**Comment by last-partizan** [comment]
Is there any news on this?

Can i donate money specifically to request solving this problem?

**Comment by brendan8c** [comment]
The extension has changed.
Now everything suits me.
Now there is a list to which I add sites that need a dark theme. I'm glad!

**Comment by last-partizan** [comment]
There's still confusing behaviour of the "Enable by default" option.

Which uses single list of sites as "Whitelist" when it's "Enabled" or "Blacklist" when it's "Disabled", which is extremely confusing: #8911

**Comment by alexanderby** [comment]
The fix landed in version 4.9.69 last year. Unfortunately some users lost their site lists during the migration when the version of the extension was different on different computers (or phones in case of Firefox).
https://github.com/darkreader/darkreader/commit/6db4210350e133af4d3f69d8786636cc1dd0477b
https://github.com/darkreader/darkreader/commit/48f3b2f533fa4e44f0b66bdddc9040e889694a1d
https://github.com/darkreader/darkreader/commit/d3ce1a572e101a545163f444e55058a4065ca5bf
https://github.com/darkreader/darkreader/commit/ca41473a04ab0eaf733c959c4600dd5674ca2a53
https://github.com/darkreader/darkreader/commit/721a2558a1b0c2f7bbb44d2908110de4b92aaf5e
https://github.com/darkreader/darkreader/commit/aef59b3af1f0e3e0d57bdb0de712280f4583708e
